src/auth/utils/timeout-functions.ts

```diff
@@ -0,0 +1,36 @@
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { UserRepository } from '../../user/user.repository';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class AccountActivationTimeouts {
+  constructor(
+    private readonly schedulerRegistry: SchedulerRegistry,
+    private readonly userRepository: UserRepository,
+  ) {}
+
+  getName(userId: number): string {
+    return `account-activation-${userId}`;
+  }
+
+  async addTimeout(
+    userId: number,
+    milliseconds: number,
+    name: string,
+  ): Promise<void> {
+    const callback = async () => {
+      await this.userRepository.removeUserById(userId);
+    };
+
+    const timeout = setTimeout(callback, milliseconds);
+    this.schedulerRegistry.addTimeout(name, timeout);
+  }
+
+  deleteTimeout(name: string): void {
+    this.schedulerRegistry.deleteTimeout(name);
+  }
+
+  getAllTimeouts(): string[] {
+    return this.schedulerRegistry.getTimeouts();
+  }
```

Comment:
If this is using `.setTimeout` under the hood then I think we should not use this. Remember our discussion about what happens when you restart the service? Unless I'm missing something and you have addressed that. But I think we should use cron instead. I assume that the account activation time is somewhere around 24h.